music-playlist-creator/script.js

```diff
@@ -66,4 +136,24 @@ function populatePlaylistCardSection(){
   }   
 }
 
+
 populatePlaylistCardSection();
+
+/* BELOW Logic WILL HANDLE SHUFFLING THE SONGS */
+
+const shuffleBtn = document.querySelector(".shuffle-btn"); //get the shuffleBtn so we can add an eventlistener to it
+
+//Function has inside is an event listener so that when shuffle is clicked songs are shuffled by using sort to randomize actual data first clearing whats on the front end and re rendering it with helper function
+function randomizeSongs(){
+
+  shuffleBtn.addEventListener("click", () =>{
+    
+    currentPlaylistData.songs.sort(() => Math.random() - 0.5);
+    songList.innerHTML = "";
+    currentPlaylistData.songs.forEach((song) => {
+      createSongBanner(song);
+    })
+  })
+}
+
+randomizeSongs(); 
```

Comment:
Do you need to wrap the listener inside the function? Can't you directly add the listener?
```suggestion
shuffleBtn.addEventListener("click", () =>{
  currentPlaylistData.songs.sort(() => Math.random() - 0.5);
  songList.innerHTML = "";
  currentPlaylistData.songs.forEach((song) => {
    createSongBanner(song);
  })
})
```

Reply:
You are very right. I only did that because I thought it was better practice. Changing it now didnt realize it would be redundant.